src/OnboardingSPA/components/NewfoldInterfaceSkeleton/SiteBuild/index.js

```diff
@@ -403,6 +405,7 @@ const SiteBuild = () => {
 
 	// Track Migration Initated Event in the Migration Step.
 	useEffect( () => {
+		fetchWPSettings().then( ( res ) => ( setIsMfeMigrationInitiated( res.nfd_migrate_site ) ) );
```

Comment:
Can we not emit the use of a state variable `isMfeMigrationInitiated` and directly call trackInstaWpMigrationEvent with a value as a param and use that to determine if `variable === true ? ACTION_MFE_MIGRATION_INITIATED: ACTION_MIGRATION_INITIATED ` ?

I think using state variables to do one-time update is an overhead as state variables may take time to reflect.